src/services/products.js

```diff
@@ -0,0 +1,102 @@
+const { Product } = require('../models');
+const { productImageService } = require('../uploads');
+
+class ProductsService {
+  async createProduct (req, res) {
+    await productImageService.uploadImage(req);
+    const product = await new Product(req.body).save();
+
+    return res.json(product);
+  }
+
+  async updateProduct (req, res) {
+    const available = await Product.findById(req.params.id).exec();
+
+    if (req.files && available) {
+      if (available.image.length !== 0) {
+        await productImageService.deleteImage(available.image);
+      }
+
+      await productImageService.uploadImage(req);
+    }
+
+    const product = await Product.findByIdAndUpdate(
+      req.params.id,
+      req.body
+    ).exec();
+
+    return product ? res.json(true) : res.boom.notFound();
+  }
+
+  async getProducts (req, res) {
+    const { _id, name, category, sizes, status, minPrice, maxPrice } =
+            (typeof req.query !== 'undefined' && req.query) || {};
+    const query = {};
+
+    if (_id != null) {
+      query._id = _id;
+    }
+    if (name != null) {
+      const manyWords = name.split(' ');
+      let regexBuilder = '';
+
+      if (manyWords.length > 1) {
+        for (const word of manyWords) {
+          regexBuilder === ''
+            ? (regexBuilder += `${word}`)
+            : (regexBuilder += `|${word}`);
+        }
+      } else {
+        regexBuilder = name;
+      }
+
+      const regex = new RegExp(regexBuilder, 'mig');
+      query.name = { $regex: regex };
+    }
+    if (category != null) {
+      query.category = category;
+    }
+    if (sizes != null) {
+      query.sizes = sizes;
+    }
+    if (status != null) {
+      query.status = status;
+    }
+    if (+Math.abs(minPrice)) {
+      query.price = { $gte: minPrice };
+    }
+    if (+Math.abs(maxPrice)) {
+      query.price = { $lte: maxPrice };
+    }
+
+    // I added the following condition to allow a user to search for products priced between amount
```

Comment:
Remove this comment. You can just answer my github comments, no necessity to add comment to code) Besides, don't close my previous comments (don't resolve them), just write "fixed" or "done" below, because it helps me to understand where were my previous comments) Instead of this check you can do next:
      query.price = { $gte: +Math.abs(minPrice) || 0 };
      
    if (+Math.abs(maxPrice)) {
      query.price['$lte'] = maxPrice;
    }

Reply:
```
TypeError: Cannot set properties of undefined (setting '$gte')
```
Trying to access 'query.price' returns undefined because the 'price' key does not exist in the object.

https://bobbyhadz.com/blog/javascript-cannot-set-property-of-undefined